config/src/config/mod.rs

```diff
@@ -94,6 +94,17 @@ pub struct BaseConfig {
     data_dir: PathBuf,
     pub role: RoleType,
     pub waypoint: WaypointConfig,
+    genesis_waypoint: Option<WaypointConfig>,
```

Comment:
also we should probably put a comment somewhere around here that distinguishes these two, it might not be clear from what you've been told thus far (maybe inferable from the task) but we can chat about it in more depth.